Sources/Algorithms/Chunked.swift

```diff
@@ -246,3 +246,136 @@ extension Collection {
     try chunked(on: projection, by: ==)
   }
 }
+
+//===----------------------------------------------------------------------===//
+// chunks(of:)
+//===----------------------------------------------------------------------===//
+
+/// A collection that presents the elements of its base collection
+/// in `SubSequence` chunks of any given size.
+///
+/// A ChunkedCollection is a lazy view on the base Collection, but it does not implicitly confer
+/// laziness on algorithms applied to its result.  In other words, for ordinary collections `c`:
+///
+/// * `c.chunks(of: 3)` does not create new storage
+/// * `c.chunks(of: 3).map(f)` maps eagerly and returns a new array
+/// * `c.lazy.chunks(of: 3).map(f)` maps lazily and returns a `LazyMapCollection`
+public struct ChunkedCollection<Base: Collection> {
```

Comment:
I’d like this name and the other chunking collection to work together a little better. As is, they would be `LazyChunked` and `ChunkedCollection`, which doesn’t really distinguish either. It’s a bit easier to add the semantics of this one to the name — what do you think about `struct ChunkedByCount<Base: ...>`?

Reply:
I like it, it takes away the possible ambiguity